src/connection_string.ts

```diff
@@ -1223,9 +1223,17 @@ export const OPTIONS = {
   pfx: { type: 'any' },
   secureProtocol: { type: 'any' },
   index: { type: 'any' },
-  // Legacy Options, these are unused but left here to avoid errors with CSFLE lib
-  useNewUrlParser: { type: 'boolean' } as OptionDescriptor,
-  useUnifiedTopology: { type: 'boolean' } as OptionDescriptor,
+  // Legacy options from v3 era
+  useNewUrlParser: {
+    type: 'boolean',
+    deprecated:
+      'useNewUrlParser has no effect since Node.js Driver version 4.0.0 and will be removed in the next major version'
+  } as OptionDescriptor,
```

Comment:
just curious - how come we need to cast `as OptionDescriptor` here?

Reply:
Because the key doesn't appear in MongoClientOptions it doesn't get typed here b/c OPTIONS is typed: `Record<keyof MongoClientOptions, OptionDescriptor>`